File: common/src/main/java/org/sandboxpowered/silica/fluid/SilicaFluidState.java
package org.sandboxpowered.silica.fluid;

import com.google.common.collect.ImmutableMap;
import org.sandboxpowered.api.fluid.Fluid;
import org.sandboxpowered.api.state.FluidState;
import org.sandboxpowered.api.state.property.Property;
import org.sandboxpowered.silica.state.BaseState;

public class SilicaFluidState extends BaseState<Fluid, FluidState> implements FluidState {
    public SilicaFluidState(Fluid base, ImmutableMap<Property<?>, Comparable<?>> properties) {
        super(base, properties);
    }

    @Override
    public Fluid getFluid() {
        return base;
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/state/fluid/SilicaFluidState.kt
package org.sandboxpowered.silica.state.fluid

import com.google.common.collect.ImmutableMap
import org.sandboxpowered.api.fluid.Fluid
import org.sandboxpowered.api.world.state.FluidState
import org.sandboxpowered.api.world.state.Property
import org.sandboxpowered.silica.state.BaseState

class SilicaFluidState(base: Fluid, properties: ImmutableMap<Property<*>, Comparable<*>>) :
    BaseState<Fluid, FluidState>(base, properties), FluidState {
    override fun getFluid(): Fluid {
        return base
    }
}